### PR TITLE
api for field format docs

### PR DESCRIPTION
**Pull request by jywadhwani**

### CODE REVIEW

File: wherehows-dao/src/main/java/wherehows/dao/view/DataTypesViewDao.java [hunk 1]
```diff
@@ -37,4 +37,13 @@ public DataTypesViewDao(EntityManagerFactory factory) {
   public List<Map<String, Object>> getAllPlatforms() throws Exception {
     throw new UnsupportedOperationException("Operation not implemented");
   }
+
+  /**
+   * Get docs for all field format types
+   */
+  public Map<String, String> getDocsForAllFieldFormats() throws Exception {
```
Comment (alyiwang) [inline wherehows-dao/src/main/java/wherehows/dao/view/DataTypesViewDao.java]: Can remove the mentioning of Docs, just  getAllFieldFormats() should be fine. 
Also, in the java docs, put a few words to explain the Map key and value

File: wherehows-frontend/app/controllers/api/v2/Dataset.java [hunk 2]
```diff
@@ -144,6 +144,16 @@ private static AclDao initAclDao() {
     }
   }
 
+  public static Promise<Result> getFieldFormatDocs() {
+    try {
+      return Promise.promise(() -> ok(
+          Json.newObject().set("fieldFormatDocs", Json.toJson(DATA_TYPES_DAO.getDocsForAllFieldFormats()))));
```
Comment (alyiwang) [inline wherehows-frontend/app/controllers/api/v2/Dataset.java]: Same as above, maybe remove docs

File: wherehows-frontend/conf/routes [hunk 3]
```diff
@@ -270,6 +270,8 @@ GET           /api/v2/list/platforms
 
 GET           /api/v2/list/compliance-data-types                                controllers.api.v2.Dataset.getComplianceDataTypes
 
+GET           /api/v2/list/field-format-docs                                    controllers.api.v2.Dataset.getFieldFormatDocs
```
Comment (alyiwang) [inline wherehows-frontend/conf/routes]: Same here

**Comment by jywadhwani** [comment]
I have made the changes to /compliance-data-types response.

**Pull request closed without merging**